rudder-lang/src/ast.rs

```diff
@@ -171,16 +172,16 @@ impl<'src> AST<'src> {
     /// Insert the variables definition into the global declaration space
     fn add_variables(&mut self, variable_declarations: Vec<(Token<'src>, PValue<'src>)>) {
         for (variable, value) in variable_declarations {
-            if let Err(e) = self.context.new_variable(
+            if let Err(e) = self.context.add_variable(
                 None,
                 variable,
-                Value::from_static_pvalue(value.clone()).unwrap(),
+                &Value::from_static_pvalue(value.clone()).unwrap(),
             ) {
                 self.errors.push(e);
             }
             let getter = |k| self.context.get(&k);
             match Value::from_pvalue(&self.enum_list, &getter, value) {
-                Err(e) => self.errors.push(e),
+                Err(e) => { self.errors.push(e); },
                 Ok(val) => {
                     self.variable_declarations.insert(variable, val);
```

Comment:
here too

Reply:
not this one, it needs a ';' to ignore return value